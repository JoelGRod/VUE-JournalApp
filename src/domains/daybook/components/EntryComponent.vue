<template>
  <div class="entry-container mb-3 p-2 pointer">
    <!-- Tittle -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">15</span>
      <span class="mx-1 fs-5">July</span>
      <span class="mx-2 fw-light">2021, Thursday</span>
    </div>

    <!-- Entry Description -->
    <div class="entry-description">
      Excepteur aliqua aliquip officia exercitation et aliqua deserunt aliqua
      fugiat. Consectetur nisi cupidatat cupidatat elit qui consequat cillum
      quis esse. Deserunt qui sunt amet est magna incididunt ut. Elit elit
      ullamco eu magna magna aute voluptate cillum proident nostrud aliquip.
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }

    .entry-description {
        font-size: 0.9rem;
    }
}
</style>