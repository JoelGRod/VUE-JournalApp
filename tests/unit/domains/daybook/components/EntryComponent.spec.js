import { shallowMount } from "@vue/test-utils";
import EntryComponent from "../../../../../src/domains/daybook/components/EntryComponent";

describe('Testing EntryComponent.vue', () => {

    const mockRouter = {
        push: jest.fn()
    };

    let wrapper;

    beforeEach(() => {
        wrapper.shallowMount(EntryComponent, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            },
            props: {
                entry: {
                    id: '-MiklDdpx9Q1PSkn008K',
                    date: 1630758078386,
                    picture: 'https://res.cloudinary.com/do7c3iy3j/image/upload/v1630758299/vue-tests/odv8p7rkeuiqrslqf6ff.jpg',
                    text: 'Hello world\n\nThis is a test entry\n\nCheers'
                }
            }
        });
    });

    test('should match with the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

});