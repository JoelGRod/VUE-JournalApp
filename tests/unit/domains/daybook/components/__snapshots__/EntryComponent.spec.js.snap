// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EntryComponent.vue should match with the snapshot 1`] = `
<div class="entry-container mb-3 p-2 pointer">
  <!-- Tittle -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">4</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-2 fw-light">2021, SÃ¡bado</span></div><!-- Entry Description -->
  <div class="entry-description">Hello world

    This is a test entry

    Cheers</div>
</div>
`;
